src/blockdata.rs

```diff
@@ -48,52 +60,114 @@ impl Block {
     /// time: 4
     /// length of aggPubkey: 1
     pub fn get_header_without_proof(&self) -> &[u8] {
-        &self.0[..Self::PROOF_POSISION]
+        &self.0[..Self::PROOF_POSITION]
     }
 
     pub fn hex(&self) -> String {
         hex::encode(&self.0)
     }
 
-    pub fn hash(&self) -> Result<BlockHash, Error> {
+    /// Returns hash for signing. This hash value doesn't include proof field. Actual block hash
+    /// includes proof data.
+    pub fn hash_for_sign(&self) -> Result<BlockHash, Error> {
```

Comment:
The hash to be signed is called `sighash`.

Reply:
I will change the expression in all place. Thanks!

Reply:
changed!